content/ios/query.md

```diff
@@ -106,13 +98,43 @@ SKYDatabase *privateDB = database;
 }];
 ```
 
-## Limiting and Offset
+## Complex Queries
+
+### Sorting the records
+We can sort records returned like this:
+
+```obj-c
+NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"_updated_at" ascending:NO];     // sorted by modificationDate
+query.sortDescriptors = @[sortDescriptor];     // apply the NSSortDescriptor to the query
+```
+
+`SKYQuery` utilizes `NSPredicate` to apply filtering on query results. You can use other parameters to sort your quries.
+
+### Limiting and Offset
 
-## Cached Queries
+We can limit the numbers of records returned like this:
 
-## Record Count
+```obj-c
+query.limit = 10;     // only show the top 10 records
+```
+
+We can also set an offset number to the query like this:
+
+```obj-c
+query.offset = 5;     // ignore the first 5 records
+```
+
+Now the first 5 records in the result list will be ignored. Your query result will start with the 6th record.
+
+### Record Count
 
 To get the number of all records matching a query, set the property
 `overallCount` property of `SKYQuery` to `YES`. The record count can be
 retrieved from `overallCount` property of `SKYQueryOperation` when
 `perRecordCompletionBlock` is first called.
+
+### Cached Queries
+
+Skygear provides a simple cached query mechanism for application that wants to display the data last seen before it gets new data from Skygear Server. To make use of this, you just need to provide an extra callback function to the `query` function, and Skygear will try to return an cached result before it gets the latest result from the Skygear server.
```

Comment:
- use "the cached data" instead of "the data last seen"
- "the cached result" instead of "an cached result"